meerk40t/main.py

```diff
@@ -340,10 +340,17 @@ def run():
     if args.batch:
         # If a batch file is specified it gets processed here.
         kernel_root.channel("console").watch(print)
+        unprint_console = True
         with args.batch as batch:
             for line in batch:
-                kernel_root(line.strip() + "\n")
-        kernel_root.channel("console").unwatch(print)
+                line = line.strip()
+                if line == "channel print console":
```

Comment:
This solution is too hard coded to work.

While it's not terrible, there might be other options are arguments in the channel or print commands (there currently are not) which means you can't do that. Also you could have done something in -e `channel print console` or in a batch file. This is a pretty narrow ranged bug but it might be more correct to write a local function that delegates to print and just add that and remove that within this. That way you can be sure that unwatching will remove that print. Since it's not the same print function that `channel print console` adds.

This is solidly identified but too kludgy of a solution to too narrow of a bug. The better solution might actually just to be do not watch print for executing the batch if you weren't asked to. Or, use a print delegate, such that `my_print_function not is print` and you can add multiple prints there. Or just echo the lines for the user.